### ISSUE TITLE
Volto delivers HTTP 200 when no backend server is available

### ISSUE THREAD

**Issue by wolbernd** [plone/volto#6447]
**Describe the bug**
When no Backend Server is available opening a Volto Site in a browser delivers a web page with an appropriate error message. However the error message is delivered to the browser with a HTTP statuscode 200. 

**To Reproduce**

1. Start a Volto Instance without a backend server (e.g. with `docker run --name plone6-frontend --rm -e RAZZLE_API_PATH=http://localhost:8080/Plone -d -p 3000:3000 plone/plone-frontend:latest` , make sure no application listens to Port 8080)
2. Open http://localhost:3000 in your browser
3. see error message about unreachable backend but page is returned with HTTP 200 

**Expected behavior**
The call to http://localhost:3000 should have a HTTP Statuscode indicating that there is something wrong with the website (HTTP 503 or 500). 

**Software (please complete the following information):**

- OS: Linux
- Browser: Firefox, Chrome, Curl
- Volto Version: Tested with 15.11.1 and 17.19.0 (latest plone-frontend docker image)
- Plone Version: n/a
- Plone REST API Version: n/a

**Comment by prreyanshu** [comment]
Can i work on this project  ??

**Comment by wolbernd** [comment]
@prreyanshu Yes Sorry, I hit a wall with this one and wasn't able to fix it.

**Comment by Shyam-Raghuwanshi** [comment]
hi @stevepiercy would you like to help me how to go through with this issue?

**Comment by stevepiercy** [comment]
@Shyam-Raghuwanshi sorry, I personally don't know how to resolve it, and have no time to  reproduce or work on it.

@wolbernd says they hit a wall, so I unassigned them. I also added a "confirmed" label because @davisagli gave it a thumbs up, so I assume he reproduced it. The first step would be to reproduce the issue, and work from there. Hope that helps.

**Comment by Shyam-Raghuwanshi** [comment]
I'm also able reproduce it

**Comment by Shyam-Raghuwanshi** [comment]
![Screenshot From 2025-01-16 14-10-35](https://github.com/user-attachments/assets/eea34b93-88a9-4b37-8ea5-3b65eb051d4a)

**Comment by pseudo-tensor** [comment]
I looked into this error and as it turns out in the `packages/volto/src/middleware/api.js` file, the line responsible for this error `ECONNREFUSED` here: 
https://github.com/plone/volto/blob/892a1d16ee31c73a1a319543b332a6d693a4ee2f/packages/volto/src/middleware/api.js#L300-L310

This is part of Server Side Rendering and hence any network activity here won't reflect in the network activities tab of devtools.
So that way the HTTP status code is correct because it reflects your browser fetching the home page and not hitting the backend
